src/ClientData/TrackDataTest.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+
+#include "ClientData/TrackData.h"
+
+namespace orbit_client_data {
+
+TEST(TrackData, Empty) {
+  TrackData track_data;
+  EXPECT_TRUE(track_data.GetChains().empty());
+  EXPECT_EQ(track_data.GetChain(0), nullptr);
+  EXPECT_EQ(track_data.GetChain(7), nullptr);
+  EXPECT_EQ(track_data.GetNumberOfTimers(), 0);
+  EXPECT_EQ(track_data.GetMaxTime(), std::numeric_limits<uint64_t>::min());
+  EXPECT_EQ(track_data.GetMinTime(), std::numeric_limits<uint64_t>::max());
+}
+
+TEST(TrackData, AddTimers) {
+  TrackData track_data;
+  orbit_client_protos::TimerInfo timer_info;
+  timer_info.set_start(2);
+  timer_info.set_end(5);
+
+  track_data.AddTimer(0, timer_info);
+
+  EXPECT_EQ(track_data.GetNumberOfTimers(), 1);
+  ASSERT_NE(track_data.GetChain(0), nullptr);
+  EXPECT_EQ(track_data.GetChain(1), nullptr);
+  EXPECT_EQ(track_data.GetChain(0)->size(), 1);
+
+  EXPECT_EQ(track_data.GetMaxTime(), 5);
+  EXPECT_EQ(track_data.GetMinTime(), 2);
+
+  timer_info.set_start(10);
+  timer_info.set_end(11);
+
+  track_data.AddTimer(1, timer_info);
```

Comment:
This is not a real case, this timer shouldn't be in depth == 1.

What do you think instead in adding also an additional timer from 8 to 11 at depth 0? Therefore, this could be a real case and you also can test if it is possible to have 2 timers at depth 0.

Nit: You also can split and make a test for an individual timer and another for this case.

Reply:
it is technically possible - this tests that no matter which depth timer is added to it updates min and max.

Reply:
Yes, I know, but I would prefer modify it as I said, because it will be also a valid case

Reply:
done.